src/main/java/org/spongepowered/common/data/processor/data/entity/TameableDataProcessor.java

```diff
@@ -0,0 +1,163 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.data.entity;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spongepowered.common.data.util.DataUtil.getData;
+
+import com.google.common.base.Optional;
+import com.mojang.util.UUIDTypeAdapter;
+import net.minecraft.entity.passive.EntityTameable;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableTameableData;
+import org.spongepowered.api.data.manipulator.mutable.entity.TameableData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.OptionalValue;
+import org.spongepowered.api.service.persistence.InvalidDataException;
+import org.spongepowered.common.data.DataProcessor;
+import org.spongepowered.common.data.manipulator.immutable.entity.ImmutableSpongeTameableData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeTameableData;
+import org.spongepowered.common.data.processor.common.AbstractSpongeDataProcessor;
+import org.spongepowered.common.data.value.AbstractBaseValue;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeOptionalValue;
+
+import javax.annotation.Nullable;
+import java.util.UUID;
+
+public class TameableDataProcessor extends AbstractSpongeDataProcessor<TameableData, ImmutableTameableData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof EntityTameable;
+    }
+
+    @Override
+    public Optional<TameableData> from(DataHolder dataHolder) {
+        if(dataHolder instanceof EntityTameable) {
+            final Optional<UUID> uuidOptional = getTamer(((EntityTameable) dataHolder));
```

Comment:
This should check if the entity is tamed. If the entity is not tamed, then there is no `TameableData` to retrieve.